src/snowflake/snowpark/column.py

```diff
@@ -461,65 +510,91 @@ def between(
+        return self._bin_op_rimpl(lambda ast: ast.sp_column_bit_xor, BitwiseXor, other)
+
+    def _unary_op_impl(self, property: Callable, operator: UnaryExpression) -> "Column":
+        ast = Column._create_ast(
+            property = property, 
+            copy_messages = {"col": self._ast},
+        )
+        return Column(operator(self._expression), ast = ast)
 
     def __neg__(self) -> "Column":
         """Unary minus."""
+        # TODO: Need SpColumnNeg IR entity
+        # return self._unary_op_impl(lambda ast: ast.sp_column_neg, UnaryMinus)
         return Column(UnaryMinus(self._expression))
 
     def equal_null(self, other: "Column") -> "Column":
         """Equal to. You can use this for comparisons against a null value."""
-        return Column(EqualNullSafe(self._expression, Column._to_expr(other)))
+        return self._bin_op_impl(lambda ast: ast.sp_column_equal_null, EqualNullSafe, other)
 
     def equal_nan(self) -> "Column":
         """Is NaN."""
-        return Column(IsNaN(self._expression))
+        return self._unary_op_impl(lambda ast: ast.sp_column_equal_nan, IsNaN)
 
     def is_null(self) -> "Column":
         """Is null."""
-        return Column(IsNull(self._expression))
+        return self._unary_op_impl(lambda ast: ast.sp_column_is_null, IsNull)
 
     def is_not_null(self) -> "Column":
         """Is not null."""
-        return Column(IsNotNull(self._expression))
+        return self._unary_op_impl(lambda ast: ast.sp_column_is_not_null, IsNotNull)
 
     # `and, or, not` cannot be overloaded in Python, so use bitwise operators as boolean operators
     def __and__(self, other: "Column") -> "Column":
         """And."""
-        return Column(And(self._expression, Column._to_expr(other)))
+        return self._bin_op_impl(lambda ast: ast.sp_column_and, And, other)
 
     def __rand__(self, other: "Column") -> "Column":
-        return Column(And(Column._to_expr(other), self._expression))  # pragma: no cover
+        return self._bin_op_rimpl(lambda ast: ast.sp_column_and, And, other)    # pragma: no cover
 
     def __or__(self, other: "Column") -> "Column":
         """Or."""
-        return Column(Or(self._expression, Column._to_expr(other)))
+        return self._bin_op_impl(lambda ast: ast.sp_column_or, Or, other)
 
     def __ror__(self, other: "Column") -> "Column":
-        return Column(And(Column._to_expr(other), self._expression))  # pragma: no cover
+        return self._bin_op_rimpl(lambda ast: ast.sp_column_or, Or, other)      # pragma: no cover
 
     def __invert__(self) -> "Column":
         """Unary not."""
+        # TODO: Need an SpColumnNot IR entity
+        # return self._unary_op_impl(lambda ast: ast.sp_column_not, Not)
         return Column(Not(self._expression))
 
     def _cast(self, to: Union[str, DataType], try_: bool = False) -> "Column":
+        # TODO: Update SpColumnCast IR entity with new field "try_", then uncomment
```

Comment:
how difficult would this be? I would be also to defer this work to another PR here. Ok, to have only simple column expr supported for now.

Reply:
Not very difficult, I have already made the changes and compiled the new protobuf in the PR in the monorepo with the modifications necessary: https://github.com/snowflakedb/snowflake/pull/183083

However, I will get rid of these `TODO`s in favor of a flexible approach in the `_create_ast` method which can silently skip IR entities and fields or raise an error depending on a parameter.